### PR TITLE
[#3867] Turn on partial parsing by default

### LINKED ISSUE #3867: Turn on partial parser by default
Pre-req before turning on:
- [x] https://github.com/dbt-labs/dbt/issues/3810
- [ ] https://github.com/dbt-labs/dbt/issues/3885
- [ ] https://github.com/dbt-labs/dbt/issues/2330#issuecomment-919798130

### PR DESCRIPTION
**Pull request by gshank**
resolves #3867

### Description

Turn on partial parsing by default.

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: core/dbt/flags.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "STATIC_PARSER": True,
     "WARN_ERROR": False,
     "WRITE_JSON": True,
-    "PARTIAL_PARSE": False,
+    "PARTIAL_PARSE": True,
```
Comment (jtcohen6) [inline core/dbt/flags.py:44]: 💯

File: core/dbt/parser/manifest.py [hunk 2]
```diff
@@ -557,6 +558,13 @@ def read_manifest_for_partial_parse(self) -> Optional[Manifest]:
                 # different version of dbt
                 is_partial_parseable, reparse_reason = self.is_partial_parsable(manifest)
                 if is_partial_parseable:
+                    # We don't want to have stale generated_at dates
+                    manifest.metadata.generated_at = datetime.utcnow()
+                    # or invocation_ids
+                    if dbt.tracking.active_user:
+                        manifest.metadata.invocation_id = dbt.tracking.active_user.invocation_id
+                    else:
+                        manifest.metadata.invocation_id = None
```
Comment (jtcohen6) [inline core/dbt/parser/manifest.py:567]: 👍

**Review by jtcohen6 — COMMENTED** [review]
Looks like the timestamp format switch threw off a couple integration tests. Assuming those are simple fixes, this LGTM

File: core/dbt/flags.py (hunk 1 above)
Comment (gshank) [reply core/dbt/flags.py:44]: Yeah. Some parts of the code want the Z ending, and some parts don't, apparently. I moved the kludging of 'generated_at' into a post serialization method in BaseArtifactMetadata.

**Comment by gshank** [comment]
The remaining test is because we're not handling 'vars' in partial parsing. To fix it the right way we'd have to implement one of the 'vars' solutions we've discussed.

**Comment by gshank** [comment]
Have modified the tests to use the --no-partial-parse flag. The issues with vars and env_vars are well known and on the roadmap.

**Review by nathaniel-may — APPROVED** [review]
🎉

**Pull request merged**